### ISSUE TITLE
Multi-GPU training methodology in detectron2

### ISSUE THREAD

**Issue by Kbkartik** [facebookresearch/detectron2#864]
In the documentation, MAX_ITER parameter is used to set the number of iterations needed for training. However, I usually follow the below procedure -
`One epoch is counted when (batch_size * number_iteration) >= number_data (or images)`

So, how does MAX_ITER relate to the standard equation? Is there a parameter to change the batch_size? In general, how does the training regime take place in detectron2? I wasn't able to find enough info on this in the documentation. Apologies if I missed. If not, can the documentation be updated?

**Comment by ppwwyyxx** [comment]
> Is there a parameter to change the batch_size?

`SOLVER.IMS_PER_BATCH`.

I can not understand the other questions. Questions like " how does the training regime take place" are very vague and maybe you can clarify what information you're looking for.

**Comment by Kbkartik** [comment]
@ppwwyyxx, sorry for the vagueness. Currently, I am using CascadeRCNN from Model Zoo to train on Cityscapes dataset using 2 GPUs. Although I have passed the num-gpus parameter while running plain_train_net.py, and also allocated 2 GPUs through SLURM batch script, I don't think the model is making use of both the GPUs. Hence, how can I verify whether my model is actually being run on 2 (or n) GPUs? And how does Detectron2 handle multi-gpu training, in terms of loading batch data, computing loss and gradients, and updating weights?

**Comment by ppwwyyxx** [comment]
It should be using multiple gpus if see the GPUs occupied by a python process in `nvidia-smi`.

We use `DistributedDataParallel` and please refer to pytorch documentation for how it works: https://pytorch.org/tutorials/intermediate/ddp_tutorial.html
